core/src/avm1/value.rs

```diff
@@ -483,6 +481,220 @@ impl<'gc> Value<'gc> {
     }
 }
 
+/// Converts an `f64` to a String with (hopefully) the same output as Flash AVM1.
+/// 15 digits are displayed (not including leading 0s in a decimal <1).
+/// Exponential notation is used for numbers <= 1e-5 and >= 1e15.
+/// Rounding done with ties rounded away from zero.
+/// NAN returns `"NaN"`, and infinity returns `"Infinity"`.
+#[allow(clippy::approx_constant)]
+fn f64_to_string(mut n: f64) -> Cow<'static, str> {
+    if n.is_nan() {
+        Cow::Borrowed("NaN")
+    } else if n == f64::INFINITY {
+        Cow::Borrowed("Infinity")
+    } else if n == f64::NEG_INFINITY {
+        Cow::Borrowed("-Infinity")
+    } else if n == 0.0 {
```

Comment:
This case is not really needed, because it is subsumed by the fast path for integers below. Or maybe it should be kept for performance?

Reply:
Oops, forgot about this! This was originally needed because Rust `format` prints `0.0` and `-0.0`, but not necessary anymore. I guess it is still nice to save some GC since 0 is probably a common value.